src/rust/lib/debug-scenes/src/shapes.rs

```diff
@@ -26,6 +28,20 @@ pub fn run_example_shapes() {
     mem::forget(app);
 }
 
+fn set_up_visualization_demo_frp(graph_editor:&GraphEditor) {
+    let network       = &graph_editor.frp.network;
+    let dummy_counter = Rc::new(Cell::new(1.0_f32));
+    frp::extend! { network
+        def _set_dumy_data = graph_editor.frp.inputs.set_visualization_data.map(move |node| {
+            let dc = dummy_counter.get();
+            dummy_counter.set(dc + 0.1);
+            let content = json!(format!("{}", 20.0 + 10.0 * dummy_counter.get().sin()));
+            let dummy_data = Some(visualization::Data::JSON { content });
+            node.visualization.frp.set_data.emit(dummy_data);
+        });
+    };
```

Comment:
The shape of the FRP is a little complex, right now we have:

```rust
keyboard shortcut  ->
graph_editor.inputs.set_data_for_selected_node ->
graph_editor.inputs.set_visualization_data -> 
[external] _set_dumy_data (with internal counter) ->
node.visualization.frp.set_data
```

Few things are bad here:
1. `set_data_for_selected_node` should not become a public API. If you want to keep it for tests rename it to `DEBUG_set_data_for_selected_node` or something with fixme to remove it. Basically, we will never have use cases to support it and it should not make its way to public API - and lets consider everything on master to be a public API. OF course it changes, but lets minimize the amount of work we got there.

2. I don't like going out of the network to some external definition and getting back there.

3. Naming `set_visualization_data` and `set_data_for_selected_node` - are inconsistent. I would name them `set_output_data` and `DEBUG_set_output_data_for_selected_node` instead. Im not using the word visualization here on purpose - we do not need to suggest what this data will be used for (maybe it will be used for something else in the future?)

What do you think about changing it to the following:

```rust
keyboard shortcut ->
graph_editor.inputs.DEBUG_set_data_for_selected_node (with internal counter) (this should just act as an emitter) ->
graph_editor.inputs.set_visualization_data -> 
node.visualization.frp.set_data
```

In such scenario we have FRP which is much closer to final version than what we've got not.